src/litdata/streaming/downloader.py

```diff
@@ -188,10 +194,42 @@ def download_file(self, remote_filepath: str, local_filepath: str) -> None:
 }
 
 
+_DOWNLOADERS = {
+    "s3://": "s3",
+    "gs://": "gs",
+    "azure://": "abfs",
+    "local:": "file",
+    "": "file",
+}
+
+
+class FsspecDownloader(Downloader):
+    def __init__(
+        self,
+        cloud_provider: str,
+        remote_dir: str,
+        cache_dir: str,
+        chunks: List[Dict[str, Any]],
+        storage_options: Dict | None = {},
+    ):
+        remote_dir = remote_dir.replace("local:", "")
+        super().__init__(cloud_provider, remote_dir, cache_dir, chunks, storage_options)
+
+    def download_file(self, remote_filepath: str, local_filepath: str) -> None:
+        if os.path.exists(local_filepath) or remote_filepath == local_filepath:
+            return
+        try:
+            with FileLock(local_filepath + ".lock", timeout=3):
```

Comment:
Hi @deependujha, 

I was reviewing the documentation and came across the section on transactions. I’m wondering if this approach might be a good alternative to locking. 

Would love to hear your thoughts: https://filesystem-spec.readthedocs.io/en/latest/features.html#transactions

Reply:
We can combine both. The main challenge is to avoid multiple processes to override the same chunks. This is why we have the lock.

Reply:
Another option would be store the data in the RAM and write to disk once used.